bcda/api/v1/api.go

```diff
@@ -285,6 +293,22 @@ func ServeData(w http.ResponseWriter, r *http.Request) {
 	} else {
 		http.ServeFile(w, r, fmt.Sprintf("%s/%s/%s", dataDir, jobID, fileName))
 	}
+
+	// Logging request for auditing
+	var jobIdInt, err = strconv.ParseUint(jobID, 10, 64)
+
+	if err == nil {
+		var jobKey, err = repository.GetJobKey(context.Background(), uint(jobIdInt), strings.TrimSpace(fileName))
+
+		if err == nil {
+			var authData, ok = r.Context().Value(auth.AuthDataContextKey).(auth.AuthData)
+
+			if ok {
+				log.API.Infof("ACO_ID: '%s', CMS_ID: '%s', Resource: '%s'", authData.ACOID, authData.CMSID, jobKey.ResourceType)
+			}
+		}
```

Comment:
Move this into a function to make it possible to simplify the error handling.

Reply:
Would you mind clarifying the intent here?  Should lines 298 to 310 into a separate function and simply invoke it from here?  We weren't currently doing anything with the err because appropriate responses were being returned by later logic in the request chain

Reply:
yeah sorry for the lack of clarity. If you move this into a separate function that returns the data you need and an error you can avoid the nested `if`s. This is a minor readability request more than anything.

Reply:
right so it would have things like:
```go
var jobIdInt, err = strconv.ParseUint(jobID, 10, 64)
if err != nil {
    return error, nil // or whatever the signature would be.
}
```

The signature could either return an error and the data you need, or it could simply perform the log. It will be easier to test if it returns the error and data.